### PR TITLE
Adds ability to generate terraform specific speakeasy tags in the openAPI yaml

### PR DESCRIPTION
**Pull request by mstanbCO**
This PR adds the ability to specify Terraform options on proto messages and methods. This allows us to generate `x-speakeasy-entity` and `x-speakeasy-entity-operation` tags in the openAPI.yaml file. These tags allow us to generate a Terraform provider with Speakeasy. 

This PR also adds support for specifying if a proto field should be required or not.

### CODE REVIEW

File: protos/apigw/v1/apigw.proto [hunk 1]
```diff
@@ -36,6 +60,20 @@ message Service {
   string docs_path = 2;
 }
 
+message TerraformEntity {
+  TerraformEntityMethodType type = 1;
+  string name = 2;
+}
+
+enum TerraformEntityMethodType {
```
Comment (pquerna) [inline protos/apigw/v1/apigw.proto:68]: can this be an embeded enum on the TerraformEntity message? do you see us re-using it outside this message?

File: example/bookstore/v1/bookstore.proto [hunk 2]
```diff
@@ -126,8 +142,16 @@ message Shelf {
...
 message Book {
+  option (apigw.v1.message).message_options = {
+    terraform_entity: {
+      name: "Book"
+    }
+  };
   // A unique book id.
-  int64 id = 1;
+  int64 id = 1 [(apigw.v1.field).field_options = {
+    required: true
```
Comment (pquerna) [inline example/bookstore/v1/bookstore.proto:152]: hmm. i'm confused about this.

we're saying, this field is "actually" required?

I wonder if we should enforce this instead through protoc-gen-validator annotations?

File: vendor/github.com/ductone/protoc-gen-apigw/apigw/v1/http_bridge.go [hunk 3]
```diff
@@ -49,10 +49,9 @@ func TimeoutForRequest(req *http.Request) (time.Duration, bool) {
 }
```
Comment (pquerna) [inline vendor/github.com/ductone/protoc-gen-apigw/apigw/v1/http_bridge.go:49]: this change looks like a vendoring issue?

File: protos/apigw/v1/apigw.proto (hunk 1 above)
Comment (mstanbCO) [reply protos/apigw/v1/apigw.proto]: changed in latest commit

File: vendor/github.com/ductone/protoc-gen-apigw/apigw/v1/http_bridge.go (hunk 3 above)
Comment (mstanbCO) [reply vendor/github.com/ductone/protoc-gen-apigw/apigw/v1/http_bridge.go:49]: hmmm tried go mod tidy and go mod vendor, but that doesn't seem to fix this.

File: example/bookstore/v1/bookstore.proto (hunk 2 above)
Comment (mstanbCO) [reply example/bookstore/v1/bookstore.proto:152]: Yeah, this makes it so we can indicate that a given field is required in the openAPI.yaml file. We will also probably want to eventually add some other field options for setting [these](https://speakeasyapi.dev/docs/using-speakeasy/create-terraform/extensions/#x-speakeasy-param-readonly) speakeasy tags on individual params. 

I'm not exactly sure what you are referring to when saying `protoc-gen-validator  annotations`. Where should I look to find this?

Comment (mstanbCO) [reply example/bookstore/v1/bookstore.proto:152]: After discussing this further, I think it’s probably best to follow this pattern for now. It’s not ideal that we have to go back and retroactively add the required tag, but to do it using the validator field from protoc-gen-validate seems like it wouldn’t always work for us. Just for strings we would have to be able to:
1. evaluate if a regex pattern requires at least 1 character
2. check for a min length
3. check for min bytes

Additionally, the bool type and any numeric types do not seem to have a way to indicate that they are required using the validator.

File: vendor/github.com/ductone/protoc-gen-apigw/apigw/v1/http_bridge.go (hunk 3 above)
Comment (mstanbCO) [reply vendor/github.com/ductone/protoc-gen-apigw/apigw/v1/http_bridge.go]: Fixed in latest commit

File: internal/apigw/apigw_openapi_schema.go [hunk 4]
```diff
@@ -86,8 +100,23 @@ func (sc *schemaContainer) Message(m pgs.Message, filter []string, nullable *boo
...
+		_, err := f.Extension(apigw_v1.E_Field, mext)
+		if err != nil {
+			return nil
+		}
+		if len(mext.FieldOptions) > 0 {
+			fOption := mext.FieldOptions[0]
+			if fOption.Required {
+				required = append(required, jn)
+			}
+		}
```
Comment (loganintech) [inline internal/apigw/apigw_openapi_schema.go:113]: Might be valuable to have a `getFieldOptions` just like `getMessageOptions`

File: internal/apigw/apigw_openapi.go [hunk 5]
```diff
@@ -171,6 +175,28 @@ func (module *Module) buildOperation(ctx pgsgo.Context, method pgs.Method, mt *m
 	return r, op, components, nil
 }
 
+func getTerraformEntityOperationExtension(operation *apigw_v1.Operation) string {
+	terraformEntity := ""
+	if operation.TerraformEntity == nil {
+		return ""
+	}
+	switch operation.TerraformEntity.Type {
```
Comment (loganintech) [inline internal/apigw/apigw_openapi.go:183]: Are we missing `GET` in this switch statement? Does the default case handle that?

Comment (mstanbCO) [reply internal/apigw/apigw_openapi.go:183]: Clarified with the speakeasy team and `get` is an alias for `read`, so I removed `GET` in the latest commit.

File: internal/apigw/apigw_openapi_schema.go (hunk 4 above)
Comment (mstanbCO) [reply internal/apigw/apigw_openapi_schema.go]: added in most recent commit

File: example/bookstore/v1/bookstore.proto (hunk 2 above)
Comment (loganintech) [reply example/bookstore/v1/bookstore.proto:152]: Maybe we can call this `terraform-required` or something. Or `required-spec`. IDK, it's just we put it on the api gateway marked as required and don't enforce its existence in the http handler itselt

Comment (mstanbCO) [reply example/bookstore/v1/bookstore.proto]: I changed the name and left a comment describing what it is for

**Review by loganintech — APPROVED** [review]

**Pull request merged**